### PR TITLE
Travis CI: Add build jobs for OSX builds

### PR DESCRIPTION
**Pull request by AlexandraTrifan**
Add build jobs for OSX builds. These create pkgs which can be published in a release.
Also, Python wheels are created in order to install Python bindings.

Signed-off-by: Alexandra.Trifan <Alexandra.Trifan@analog.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,6 +19,22 @@ matrix:
...
+      os: osx
+      osx_image: xcode10.1
+      env:
+        - EXTRA_SSH=-oHostKeyAlgorithms=+ssh-dss
+    - compiler: "gcc"
+      os: osx
+      osx_image: xcode11
+      env:
+        - EXTRA_SSH=-oHostKeyAlgorithms=+ssh-dss
+
```
Comment (commodo) [inline .travis.yml]: nitpick: extra line

File: .travis.yml [hunk 2]
```diff
@@ -19,6 +19,22 @@ matrix:
       env:
         - OS_TYPE=doxygen
       addons: { apt: { packages: ["cmake", "graphviz"] } }
+    - compiler: "gcc"
+      os: osx
+      osx_image: xcode9.2
+      env:
+        - EXTRA_SSH=-oHostKeyAlgorithms=+ssh-dss
```
Comment (commodo) [inline .travis.yml]: curios: are we planning to store stuff on a server?
this env-var `EXTRA_SSH` is used in the upload scripts;
depending on how/if this is done, this is needed or not;

also, I would drop xcode9.2: it's really old;
xcode10 is old-ish;

at the rate Apple is updating Xcode, we might see an Xcode 12 pretty soon

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,11 +19,11 @@
 #
 
 cmake_minimum_required(VERSION 3.1.3)
-project(libm2k LANGUAGES CXX VERSION 0.1.0)
```
Comment (commodo) [inline CMakeLists.txt]: a commit desc would be nice

**Comment by commodo** [comment]
rest looks good

File: .travis.yml (hunk 2 above)
Comment (AlexandraTrifan) [reply .travis.yml]: We'll have to store the artifacts somewhere. If it's not a problem, we can leave the "extra_ssh" there for now.

File: .travis.yml (hunk 1 above)
Comment (AlexandraTrifan) [reply .travis.yml]: removed

File: .travis.yml (hunk 2 above)
Comment (commodo) [reply .travis.yml]: Fine to leave it

**Pull request merged**